src/steamship/agents/schema/conversation_memory.py

```diff
@@ -0,0 +1,104 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import tiktoken
+from pydantic.main import BaseModel
+
+from steamship import Block
+from steamship.agents.memory.chathistory import ChatHistory
+from steamship.data.tags.tag_constants import RoleTag
+
+
+class ConversationMemory(BaseModel, ABC):
+    @abstractmethod
+    def messages(self, chat_history: ChatHistory) -> List[Block]:
+        pass
+
+    def messages_as_string(
+        self, chat_history: ChatHistory, user_prefix: str = "User", assistant_prefix: str = "AI"
+    ) -> str:
+        messages = self.messages(chat_history)
+        print(f"messages: {messages}")
+        as_strings = []
+        for block in messages:
+            role = block.chat_role
+            if role == RoleTag.USER:
+                as_strings.append(f"{user_prefix}: {block.text}")
+            elif role == RoleTag.ASSISTANT:
+                as_strings.append(f"{assistant_prefix}: {block.text}")
+            elif role == RoleTag.SYSTEM:
+                as_strings.append(f"System: {block.text}")
+            elif role == RoleTag.AGENT:
+                as_strings.append(f"Agent: {block.text}")
+        return "\n".join(as_strings)
+
+
+class NoMemory(ConversationMemory):
```

Comment:
+1 to this idea - it's great.

If I'm reading this right: the way to not have memory is to actively chose the memory type NoMemory?